hal/targets.json

```diff
@@ -1823,7 +1823,7 @@
         "inherits": ["Target"],
         "core": "Cortex-M0+",
         "default_toolchain": "uARM",
-        "supported_toolchains": ["GCC_ARM", "uARM"],
+        "supported_toolchains": ["GCC_ARM", "uARM", "IAR"],
```

Comment:
We had issues before where the "ARM" toolchain doesn't produce code that is compact enough to fit on our M0's. Maybe we should just drop M0 support from mbed 5 then?